File: remosu/test_simple_lists.py

from simple_lists import SList, DList


def test_simple_list():    
    slist = SList() 
    assert slist.find("fred") == None 
    slist.add("fred")
    assert "fred" == slist.find("fred").value()
    assert slist.find("wilma") == None 
    slist.add("wildma") 
    assert "fred" == slist.find("fred").value()
    assert "wildma" == slist.find("wildma").value()
    assert ["fred", "wildma"] == slist.values()

    slist = SList()
    slist.add("fred")
    slist.add("wilma")
    slist.add("betty")
    slist.add("barney")
    assert ["fred", "wilma", "betty", "barney"] == slist.values()
    slist.delete(slist.find("wilma"))
    print slist.values()
    assert ["fred", "betty", "barney"] == slist.values()
    slist.delete(slist.find("barney"))
    assert ["fred", "betty"] == slist.values()
    slist.delete(slist.find("fred"))
    assert ["betty"] == slist.values()
    slist.delete(slist.find("betty"))
    assert [] == slist.values()


def test_double_linked_list():    
    dlist = DList() 
    assert dlist.find("fred") == None 
    dlist.add("fred")
    assert "fred" == dlist.find("fred").value()
    assert dlist.find("wilma") == None 
    dlist.add("wildma") 
    assert "fred" == dlist.find("fred").value()
    assert "wildma" == dlist.find("wildma").value()
    assert ["fred", "wildma"] == dlist.values()

    dlist = DList()
    dlist.add("fred")
    dlist.add("wilma")
    dlist.add("betty")
    dlist.add("barney")
    assert ["fred", "wilma", "betty", "barney"] == dlist.values()
    dlist.delete(dlist.find("wilma"))
    #print dlist.values()
    assert ["fred", "betty", "barney"] == dlist.values()
    dlist.delete(dlist.find("barney"))
    assert ["fred", "betty"] == dlist.values()
    dlist.delete(dlist.find("fred"))
    assert ["betty"] == dlist.values()
    dlist.delete(dlist.find("betty"))
    assert [] == dlist.values()

    dlist.add("pepe")
    dlist.add("cuco")
    assert dlist._next._prev == dlist
    dlist.add("paco")
    assert dlist.find("cuco")._prev.value() == "pepe"
    assert dlist.find("cuco")._next.value() == "paco"
    
    dlist.delete(dlist.find("cuco"))
    assert dlist.find("cuco") is None
    assert dlist.find("paco")._prev.value() == "pepe"

File: remosu/test_simple_lists.js
var assert = require('assert');
var simple_lists = require('./simple_lists');

slist = new simple_lists.SList();
assert.equal(slist.find("fred"), null);
slist.add("fred")
assert.equal("fred", slist.find("fred").value())
assert.equal(null, slist.find("wilma"))
slist.add("wilma")
assert.equal("fred",  slist.find("fred").value())
assert.equal("wilma", slist.find("wilma").value())
assert.deepEqual(["fred", "wilma"], slist.values())

list = new simple_lists.SList();
list.add("fred")
list.add("wilma")
list.add("betty")
list.add("barney")
assert.deepEqual(["fred", "wilma", "betty", "barney"], list.values())
list.delete(list.find("wilma"))
assert.deepEqual(["fred", "betty", "barney"], list.values())
list.delete(list.find("barney"))
assert.deepEqual(["fred", "betty"], list.values())
list.delete(list.find("fred"))
assert.deepEqual(["betty"], list.values())
list.delete(list.find("betty"))
assert.deepEqual([], list.values())

